lxd/device/gpu_physical.go

```diff
@@ -349,9 +379,29 @@ func (d *gpuPhysical) Stop() (*deviceConfig.RunConfig, error) {
 	}
 
 	if d.inst.Type() == instancetype.Container {
-		err := unixDeviceRemove(d.inst.DevicesPath(), "unix", d.name, "", &runConf)
-		if err != nil {
-			return nil, err
+		v := d.volatileGet()
+		if v["cdi"] == "true" {
+			cdiID, err := cdi.ToCDI(d.config["id"])
+			if !cdiID.Empty() {
+				var formattedErr error
```

Comment:
Why the new formattedErr var, and not just overriding err when needed?

Reply:
fixed